src/cmd/apply-as-apps.ts

```diff
@@ -113,17 +120,59 @@ const removeApplication = async (release: HelmRelease): Promise<void> => {
   }
 }
 
+function getResources(values: Record<string, any>) {
+  const config = values
+  const resources: ResourceRequirements = {
+    limits: {
+      cpu: config.controller?.resources?.limits?.cpu || undefined,
+      memory: config.controller?.resources?.limits?.memory || undefined,
+    },
+    requests: {
+      cpu: config.controller?.resources?.requests?.cpu || undefined,
+      memory: config.controller?.resources?.requests?.memory || undefined,
+    },
+  }
+  return resources
+}
+
+async function patchArgocdResources(release: HelmRelease, values: Record<string, any>) {
+  if (release.name === 'argocd') {
+    if (await hasStsOOMKilledPods('argocd-application-controller', 'argocd', k8s.app(), k8s.core(), d)) {
```

Comment:
Will this check cover use case when pods are in running state and are soon going to get OOMKilled.
Wouldn't be better idea to cross check Sts resource request with the one declared in values and if it differs to provide a patch?